src/main/java/com/gluonhq/substrate/target/LinuxLinkerFlags.java

```diff
@@ -127,28 +139,28 @@
      *
      * If pkg-config fails, adds amendment instructions to missingPackages.
      */
-    private List<String> lookupPackageFlags(PkgInfo pkgInfo, List<String> missingPackages) {
+    private List<String> lookupPackageFlags(PkgInfo pkgInfo) throws IOException, InterruptedException {
         if (pkgInfo.hardwired != null) {
             return List.of(pkgInfo.hardwired);
         }
 
         String pkgName = pkgInfo.pkgName;
         ProcessRunner process = new ProcessRunner("/usr/bin/pkg-config", "--libs", pkgName);
-        try {
-            if (process.runProcess("Get config for " + pkgName) != 0) {
-                missingPackages.add(pkgInfo.installName + " (for pkgConfig " + pkgName + ")");
-                return List.of();
-            }
-        } catch (IOException | InterruptedException e) {
-            throw new IllegalStateException("Failed to lookup linker flags", e);
+        if (process.runProcess("Get config for " + pkgName) != 0) {
+            missingPackages.add(pkgInfo.installName + " (for pkgConfig " + pkgName + ")");
+            return List.of();
         }
 
         List<String> flags = List.of(process.getResponse().trim().split(" "));
         Logger.logDebug("Pkg " + pkgName + " provided flags: " + flags);
         return flags;
     }
 
-    private void printUpdateInstructionsAndFail(List<String> missingPackages) {
+    private boolean anyOsPackageIsMissing() {
```

Comment:
This method name looks odd to me. A better alternative can be `isOSPackageMissing`.